### PR TITLE
Add examples/tests. Use examples as part of documentation. Add utilit…

### PR DESCRIPTION
**Pull request by vtereso**
Added example yaml files
Add tests to ensure valid yaml
Add utility to update docs and test to keep them fresh

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -87,12 +87,11 @@ go test -v -tags=e2e -count=1 ./test -run ^TestEventListener
 
 There are no yaml tests at the moment. These tests will be created under
 [Issue #11](https://github.com/tektoncd/triggers/issues/11).
```
Comment (ncskier) [inline test/README.md]: Thanks for adding these instructions to the test README. However, can you please delete these two above lines now that we have the yaml tests 😄

Comment (vtereso) [reply test/README.md]: Whoops. Will do.

**Comment by ncskier** [comment]
Thanks Vincent!
\lgtm

**Comment by ncskier** [comment]
Thanks Vincent!
/lgtm

**Pull request merged**